### PR TITLE
Add initial changelog for 1.11.0

### PR DESCRIPTION
**Pull request by tiborvass**
This PR is carrying #21422 and opens it against the newly created bump_v1.11.0 branch.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,137 @@ information on the list of deprecated flags and APIs please have a look at
...
 be found.
 
+## 1.11.0 (2016-04-12)
+
+**IMPORTANT**: With Docker 1.11, a docker installation is now made of 4 binaries (`docker`, [`docker-containerd`](https://github.com/docker/containerd), [`docker-containerd-shim`](https://github.com/docker/containerd) and [`docker-runc`](https://github.com/opencontainers/runc)). If you have scripts relying on docker being a single static binaries, please make sure to update them. Interaction with the daemon stay the same otherwise, the usage of the other binaries should be transparent.
+
+### Builder
+
+- Fix a bug where Docker would not used the correct uid/gid when processing the `WORKDIR` command ([#21033](https://github.com/docker/docker/pull/21033))
+- Fix a bug where copy operations would not use the proper uid/gid ([#20782](https://github.com/docker/docker/pull/20782), [#21162](https://github.com/docker/docker/pull/21162))
```
Comment (thaJeztah) [inline CHANGELOG.md]: Should we mention that these issues were when using user-namespaces, or would that be too much information?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,137 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Client
+
+* Usage of the `:` separator for security option has been deprecated. `=` should be used instead ([#21232](https://github.com/docker/docker/pull/21232))
++ The client user agent is now passed to the registry on `pull`, `build`, `push`, `login` and `search` operations ([#21306](https://github.com/docker/docker/pull/21306), [#21373](https://github.com/docker/docker/pull/21373))
```
Comment (thaJeztah) [inline CHANGELOG.md]: Should this be below "Distribution"?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,137 @@ information on the list of deprecated flags and APIs please have a look at
...
+**IMPORTANT**: With Docker 1.11, a docker installation is now made of 4 binaries (`docker`, [`docker-containerd`](https://github.com/docker/containerd), [`docker-containerd-shim`](https://github.com/docker/containerd) and [`docker-runc`](https://github.com/opencontainers/runc)). If you have scripts relying on docker being a single static binaries, please make sure to update them. Interaction with the daemon stay the same otherwise, the usage of the other binaries should be transparent.
+
+### Builder
+
+- Fix a bug where Docker would not used the correct uid/gid when processing the `WORKDIR` command ([#21033](https://github.com/docker/docker/pull/21033))
+- Fix a bug where copy operations would not use the proper uid/gid ([#20782](https://github.com/docker/docker/pull/20782), [#21162](https://github.com/docker/docker/pull/21162))
+
+### Client
+
+* Usage of the `:` separator for security option has been deprecated. `=` should be used instead ([#21232](https://github.com/docker/docker/pull/21232))
```
Comment (thaJeztah) [inline CHANGELOG.md]: It's also mentioned in the deprecation docs, so may be redundant here (but I'm good with keeping it in both)

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,137 @@ information on the list of deprecated flags and APIs please have a look at
 https://docs.docker.com/misc/deprecated/ where target removal dates can also
```
Comment (thaJeztah) [inline CHANGELOG.md]: This URL changed to https://docs.docker.com/engine/deprecated/

File: CHANGELOG.md (hunk 2 above)
Comment (tiborvass) [inline CHANGELOG.md]: i think its fine in client, but i dont mind either way

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,137 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix a bug where Docker would not used the correct uid/gid when processing the `WORKDIR` command ([#21033](https://github.com/docker/docker/pull/21033))
+- Fix a bug where copy operations would not use the proper uid/gid ([#20782](https://github.com/docker/docker/pull/20782), [#21162](https://github.com/docker/docker/pull/21162))
+
+### Client
+
+* Usage of the `:` separator for security option has been deprecated. `=` should be used instead ([#21232](https://github.com/docker/docker/pull/21232))
++ The client user agent is now passed to the registry on `pull`, `build`, `push`, `login` and `search` operations ([#21306](https://github.com/docker/docker/pull/21306), [#21373](https://github.com/docker/docker/pull/21373))
+* Value passed to `--hostname` are now treated as FQDN ([#20200](https://github.com/docker/docker/pull/20200))
+* Docker info will now warn users if it can not detect the kernel version or the operating system ([#21128](https://github.com/docker/docker/pull/21128))
+- Fix an issue where one letter directory name could not be used as source for volumes ([#21106](https://github.com/docker/docker/pull/21106))
```
Comment (thaJeztah) [inline CHANGELOG.md]: Move to "volumes" section?

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,137 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix a bug where credentials were retained after a user logging out ([#20860](https://github.com/docker/docker/pull/20860))
```
Comment (thaJeztah) [inline CHANGELOG.md]: This was a consequence of the new external credentials store, so wasn't in 1.10; I think we can remove this

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,9 +2,140 @@
...
+- Fix a file descriptor leak that would occur every time plugins were enumerated ([#20686](https://github.com/docker/docker/pull/20686))
+- Fix an issue where Authz plugin would corrupt the payload body when faced with a large amount of data ([#20602](https://github.com/docker/docker/pull/20602))
+
+### Runtime
+
++ It is now possible for containers to share the NET and IPC namespaces when `userns` is enabled ([#21383](https://github.com/docker/docker/pull/21383))
++ `docker inspect <image-id>` will now expose the rootfs layers ([#21370](https://github.com/docker/docker/pull/21370))
++ Docker Windows gained a minimal `top` implementation ([#21354](https://github.com/docker/docker/pull/21354))
+* Docker learned to report the faulty exe when a container cannot be started due to its condition ([#21345](https://github.com/docker/docker/pull/21345))
++ `docker run -v` now accepts a new flag `nocopy`. This tell the runtime not to copy the container path content into the volume (which is the default behavior) ([#21223](https://github.com/docker/docker/pull/21223))
```
Comment (thaJeztah) [inline CHANGELOG.md]: Probably better in the "volumes" section

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,9 +2,140 @@
...
++ Docker now allows setting a container hostname via the `--hostname` flag ([#20177](https://github.com/docker/docker/pull/20177))
++ Docker now allows executing privileged container while running with `--userns-remap` if both `--privileged` and the new `--userns=host` flag are specified ([#20111](https://github.com/docker/docker/pull/20111))
++ Docker now supports external credential stores ([#20107](https://github.com/docker/docker/pull/20107))
+- Fix Docker not cleaning up correctly old containers upon restarting after a crash ([#19679](https://github.com/docker/docker/pull/19679))
+* Docker will now error out if it doesn't recognize a configuration key within the config file ([#19517](https://github.com/docker/docker/pull/19517))
+- Fix container loading, on daemon startup, when they depends on a plugin running within a container ([#19500](https://github.com/docker/docker/pull/19500))
+- Fix a panic that would occur when failing to load a layer on store instantiation ([#19458](https://github.com/docker/docker/pull/19458))
+- Fix a panic that occurred when a `exec` was started multiple times ([#19369](https://github.com/docker/docker/pull/19369))
```
Comment (thaJeztah) [inline CHANGELOG.md]: I think this was in 1.10 https://github.com/docker/docker/pull/19369

File: CHANGELOG.md [hunk 9]
```diff
@@ -2,9 +2,140 @@
...
+* `docker update` learned how to change a container restart policy ([#19116](https://github.com/docker/docker/pull/19116))
+* `docker inspect` now also returns a new `State` field containing the container state in a human readable way (i.e. one of `created`, `restarting`, `running`, `paused`, `exited` or `dead`)([#18966](https://github.com/docker/docker/pull/18966))
```
Comment (thaJeztah) [inline CHANGELOG.md]: s/human/machine/

File: CHANGELOG.md [hunk 10]
```diff
@@ -2,9 +2,140 @@
...
+- Fix docker configuration reloading to only alter value present in the given config file ([#20604](https://github.com/docker/docker/pull/20604))
++ Docker now allows setting a container hostname via the `--hostname` flag ([#20177](https://github.com/docker/docker/pull/20177))
++ Docker now allows executing privileged container while running with `--userns-remap` if both `--privileged` and the new `--userns=host` flag are specified ([#20111](https://github.com/docker/docker/pull/20111))
++ Docker now supports external credential stores ([#20107](https://github.com/docker/docker/pull/20107))
+- Fix Docker not cleaning up correctly old containers upon restarting after a crash ([#19679](https://github.com/docker/docker/pull/19679))
+* Docker will now error out if it doesn't recognize a configuration key within the config file ([#19517](https://github.com/docker/docker/pull/19517))
+- Fix container loading, on daemon startup, when they depends on a plugin running within a container ([#19500](https://github.com/docker/docker/pull/19500))
+- Fix a panic that would occur when failing to load a layer on store instantiation ([#19458](https://github.com/docker/docker/pull/19458))
```
Comment (thaJeztah) [inline CHANGELOG.md]: This was in 1.10 https://github.com/docker/docker/pull/19458

File: CHANGELOG.md [hunk 11]
```diff
@@ -2,9 +2,140 @@
...
++ Docker now relies on [`containerd`](https://github.com/docker/containerd) and [`runc`](https://github.com/opencontainers/runc) to spawn containers. ([#20662](https://github.com/docker/docker/pull/20662))
+- Fix docker configuration reloading to only alter value present in the given config file ([#20604](https://github.com/docker/docker/pull/20604))
++ Docker now allows setting a container hostname via the `--hostname` flag ([#20177](https://github.com/docker/docker/pull/20177))
++ Docker now allows executing privileged container while running with `--userns-remap` if both `--privileged` and the new `--userns=host` flag are specified ([#20111](https://github.com/docker/docker/pull/20111))
++ Docker now supports external credential stores ([#20107](https://github.com/docker/docker/pull/20107))
+- Fix Docker not cleaning up correctly old containers upon restarting after a crash ([#19679](https://github.com/docker/docker/pull/19679))
+* Docker will now error out if it doesn't recognize a configuration key within the config file ([#19517](https://github.com/docker/docker/pull/19517))
```
Comment (thaJeztah) [inline CHANGELOG.md]: This was in 1.10 https://github.com/docker/docker/pull/19517

File: CHANGELOG.md (hunk 8 above)
Comment (tiborvass) [inline CHANGELOG.md]: Confirmed: https://github.com/docker/docker/commit/81cd580d68032b5690898728b9862e5dadac6e9c

File: CHANGELOG.md [hunk 12]
```diff
@@ -2,9 +2,140 @@
...
+* The devmapper driver learned the `dm.min_free_space` option. If the mapped device free space reaches the passed value, new device creation will be prohibited. ([#20786](https://github.com/docker/docker/pull/20786))
++ Docker can now prevent processes in container to gain new privileges via the `--security-opt=no-new-privileges` flag ([#20727](https://github.com/docker/docker/pull/20727))
+- Starting a container with the `--device` option will now correctly resolves symlinks ([#20684](https://github.com/docker/docker/pull/20684))
++ Docker now relies on [`containerd`](https://github.com/docker/containerd) and [`runc`](https://github.com/opencontainers/runc) to spawn containers. ([#20662](https://github.com/docker/docker/pull/20662))
+- Fix docker configuration reloading to only alter value present in the given config file ([#20604](https://github.com/docker/docker/pull/20604))
++ Docker now allows setting a container hostname via the `--hostname` flag ([#20177](https://github.com/docker/docker/pull/20177))
```
Comment (thaJeztah) [inline CHANGELOG.md]: The new thing here is that it's now possible to do this when `--net=host` is used

File: CHANGELOG.md [hunk 13]
```diff
@@ -25,9 +156,9 @@ be found.
 
 ### Security
 
-- Fix linux32 emulation to fail during docker build [#20672](https://github.com/docker/docker/pull/20672)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Oh, think the whitespace here and below got lost; these were deliberate to format the output

File: CHANGELOG.md [hunk 14]
```diff
@@ -2,9 +2,137 @@
...
++ The fluentd log driver learned the following options: `fluentd-address`, `fluentd-buffer-limit`, `fluentd-retry-wait`, `fluentd-max-retries` and `fluentd-async-connect` ([#19439](https://github.com/docker/docker/pull/19439))
++ Docker learned to send log to Google Cloud via the new `gcplogs` logging driver. ([#18766](https://github.com/docker/docker/pull/18766))
+
+
+### Misc
+
++ When saving linked images together with `docker save` a subsequent `docker load` will correctly restore their parent/child relationship ([#21385](https://github.com/docker/docker/pull/c))
++ Support for building the Docker cli for OpenBSD was added ([#21325](https://github.com/docker/docker/pull/21325))
++ Labels can now be applied at network, volume and image creation ([#21270](https://github.com/docker/docker/pull/21270))
+* The `dockremap` is now created as a system user ([#21266](https://github.com/docker/docker/pull/21266))
```
Comment (kencochrane) [inline CHANGELOG.md]: typo? dockermap?

Comment (thaJeztah) [inline CHANGELOG.md]: no the user/group is called like this (for "remap" user)

File: CHANGELOG.md [hunk 15]
```diff
@@ -2,9 +2,137 @@
...
+### Builder
+
+- Fix a bug where Docker would not used the correct uid/gid when processing the `WORKDIR` command ([#21033](https://github.com/docker/docker/pull/21033))
+- Fix a bug where copy operations with userns would not use the proper uid/gid ([#20782](https://github.com/docker/docker/pull/20782), [#21162](https://github.com/docker/docker/pull/21162))
+
+### Client
+
+* Usage of the `:` separator for security option has been deprecated. `=` should be used instead ([#21232](https://github.com/docker/docker/pull/21232))
++ The client user agent is now passed to the registry on `pull`, `build`, `push`, `login` and `search` operations ([#21306](https://github.com/docker/docker/pull/21306), [#21373](https://github.com/docker/docker/pull/21373))
+* Value passed to `--hostname` are now treated as FQDN ([#20200](https://github.com/docker/docker/pull/20200))
```
Comment (sanimej) [inline CHANGELOG.md]: Change  ``Value passed to `--hostname` are now treated as FQDN`` to `Allow setting the Domainname and Hostname separately through the API`

File: CHANGELOG.md (hunk 14 above)
Comment (kencochrane) [inline CHANGELOG.md]: cool, I just wanted to make sure, because I don't know how many times a day I type dockre instead of docker :)

File: CHANGELOG.md [hunk 16]
```diff
@@ -2,9 +2,137 @@
...
+
+### Networking
+
+- Fix a bug where IPV6 addresses were not properly handled ([#20842](https://github.com/docker/docker/pull/20842))
+* `docker network inspect` will now report all endpoints whether they have an active container or not ([#21160](https://github.com/docker/docker/pull/21160))
++ Experimental support for the MacVlan and IPVlan network drivers have been added ([#21122](https://github.com/docker/docker/pull/21122))
+* Output of `docker network ls` is now sorted by network name ([#20383](https://github.com/docker/docker/pull/20383))
+- Fix a bug where Docker would allow a network to be created with the reserved `default` name ([#19431](https://github.com/docker/docker/pull/19431))
+* `docker network inspect` now returns whether a network is internal or not ([#19357](https://github.com/docker/docker/pull/19357))
++ Docker learned to create ipv6 enabled networks (`docker network create --ipv6`). This show up as a new  `EnableIPv6` field in `docker inspect`. ([#17513](https://github.com/docker/docker/pull/17513))
```
Comment (aboch) [inline CHANGELOG.md]: Can we rephrase this as

Control ipv6 via explicit option when creating a network (`docker network create --ipv6`). This show up as a new  `EnableIPv6` field in `docker inspect`. ([#17513](https://github.com/docker/docker/pull/17513))

Comment (tiborvass) [inline CHANGELOG.md]: @aboch you mean in `docker network inspect`

File: CHANGELOG.md [hunk 17]
```diff
@@ -2,9 +2,137 @@
...
+* Docker info now reports the value of cgroup KernelMemory or emits a warning if it is not supported ([#20863](https://github.com/docker/docker/pull/20863))
+* Docker info now also reports the cgroup driver in use ([#20388](https://github.com/docker/docker/pull/20388))
+* Docker completion is now available on PowerShell ([#19894](https://github.com/docker/docker/pull/19894))
+* `dockerinit` is no more ([#19490](https://github.com/docker/docker/pull/19490),[#19851](https://github.com/docker/docker/pull/19851))
++ Support for building Docker on arm64 was added ([#19013](https://github.com/docker/docker/pull/19013))
++ Experimental support for building docker.exe in a native Windows Docker installation ([#18348](https://github.com/docker/docker/pull/18348))
+
+### Networking
+
+- Fix a bug where IPv6 addresses were not properly handled ([#20842](https://github.com/docker/docker/pull/20842))
```
Comment (sanimej) [inline CHANGELOG.md]: We don't support overlay network over IPv6. This should be moved to runtime and probably change the description to `Handle IPv6 peers correctly in neighbor discovery`

File: CHANGELOG.md [hunk 18]
```diff
@@ -2,9 +2,137 @@
...
+### Distribution
+
+- Fix a panic that occurred when pulling an images with 0 layers ([#21222](https://github.com/docker/docker/pull/21222))
+- Fix a panic that could occur on error while pushing to a registry with a misconfigured token service ([#21212](https://github.com/docker/docker/pull/21212))
++ All first-level delegation roles are now signed when doing a trusted push ([#21046](https://github.com/docker/docker/pull/21046))
++ OAuth support for registries was added ([#20970](https://github.com/docker/docker/pull/20970))
+* `docker login` now handles token using the implementation found in [docker/distribution](https://github.com/docker/distribution) ([#20832](https://github.com/docker/docker/pull/20832))
+* `docker login` will no longer prompt for an email ([#20565](https://github.com/docker/docker/pull/20565))
+* Docker will now fallback to registry V1 if no basic auth credentials are available ([#20241](https://github.com/docker/docker/pull/20241))
+* Docker will now try to resume layer download where it left timeout after a network error/timeout ([#19840](https://github.com/docker/docker/pull/19840))
```
Comment (dmcgowan) [inline CHANGELOG.md]: "where it left off"

**Comment by mavenugo** [comment]
@tiborvass @sanimej @aboch thanks for fixing the changelog. we do have a few more updates to it. But given that RC1 is blocked by it, I think we should proceed now with the RC1 and fix the changelog in parallel.

LGTM

**Comment by thaJeztah** [comment]
LGTM

File: CHANGELOG.md [hunk 19]
```diff
@@ -2,9 +2,137 @@
...
++ Docker can now prevent processes in container to gain new privileges via the `--security-opt=no-new-privileges` flag ([#20727](https://github.com/docker/docker/pull/20727))
+- Starting a container with the `--device` option will now correctly resolves symlinks ([#20684](https://github.com/docker/docker/pull/20684))
++ Docker now relies on [`containerd`](https://github.com/docker/containerd) and [`runc`](https://github.com/opencontainers/runc) to spawn containers. ([#20662](https://github.com/docker/docker/pull/20662))
+- Fix docker configuration reloading to only alter value present in the given config file ([#20604](https://github.com/docker/docker/pull/20604))
++ Docker now allows setting a container hostname via the `--hostname` flag when `--net=host` ([#20177](https://github.com/docker/docker/pull/20177))
++ Docker now allows executing privileged container while running with `--userns-remap` if both `--privileged` and the new `--userns=host` flag are specified ([#20111](https://github.com/docker/docker/pull/20111))
++ Docker now supports external credential stores ([#20107](https://github.com/docker/docker/pull/20107))
```
Comment (dmcgowan) [inline CHANGELOG.md]: Does this make more sense under client? Possibly highlight that these are registry credentials as well

**Comment by dmcgowan** [comment]
LGTM

File: CHANGELOG.md [hunk 20]
```diff
@@ -2,9 +2,137 @@
 
 Items starting with `DEPRECATE` are important deprecation notices. For more
 information on the list of deprecated flags and APIs please have a look at
-https://docs.docker.com/misc/deprecated/ where target removal dates can also
+https://docs.docker.com/engine/deprecated/ where target removal dates can also
 be found.
 
+## 1.11.0 (2016-04-12)
+
+**IMPORTANT**: With Docker 1.11, a docker installation is now made of 4 binaries (`docker`, [`docker-containerd`](https://github.com/docker/containerd), [`docker-containerd-shim`](https://github.com/docker/containerd) and [`docker-runc`](https://github.com/opencontainers/runc)). If you have scripts relying on docker being a single static binaries, please make sure to update them. Interaction with the daemon stay the same otherwise, the usage of the other binaries should be transparent.
```
Comment (jhowardmsft) [inline CHANGELOG.md]: Can we do an update on this. Something like "With Docker 1.11, a `Linux` docker installation blah blah blah....". Then add a sentence at the end "A `Windows` docker installation remains a single binary".

Comment (thaJeztah) [inline CHANGELOG.md]: @jhowardmsft we want to cleanup the change log a bit for the next RC. If you see a chance to complete the "blah blah", perhaps we can update the release notes on GitHub in the meantime

Comment (jhowardmsft) [inline CHANGELOG.md]: I did :) https://github.com/docker/docker/pull/21450
